pinot-core/src/main/java/org/apache/pinot/core/common/ObjectSerDeUtils.java

```diff
@@ -474,21 +477,31 @@ public byte[] serialize(HyperLogLog hyperLogLog) {
     @Override
     public HyperLogLog deserialize(byte[] bytes) {
       try {
-        return HyperLogLog.Builder.build(bytes);
-      } catch (IOException e) {
-        throw new RuntimeException("Caught exception while de-serializing HyperLogLog", e);
+        IntBuffer intBuf =
+                ByteBuffer.wrap(bytes)
```

Comment:
Since we need to wrap the bytes, let's implement `deserialize(ByteBuffer byteBuffer)` to avoid the unnecessary bytes copying

Reply:
Good idea, fixed this